src/secrets/CommandEnv.ts

```diff
@@ -160,7 +157,27 @@ class CommandEnv extends CommandPolykey {
                     utils.never();
                 }
               }
-              envp[newName] = secretContent;
+
+              // Find if we need to preserve the newline for this secret
+              let preserveNewline = false;
+              if (options.preserveNewline) {
+                for (const pair of options.preserveNewline) {
+                  if (
+                    pair[0] === nameOrId &&
+                    (pair[1] === newName || pair[1] == null)
+                  ) {
+                    preserveNewline = true;
+                    break;
+                  }
+                }
+              }
```

Comment:
> Secondly, if you're checking if something is within a set, you need to use a map rather than iterate over the whole set. I think you're checking if a path or a whole vault for that path is preserving the newline here? That should have a comment explaining that as well. But for the most part rather than iterate over the whole set of `options.preserveNewLine` You should structure it as a map and use that instead. To check if something is in a map is log complexity vs the linear complexity of iterating over a list.

I'm not sure how to achieve this. `options.preserveNewline` is an array of parsed environment paths like the ones parsed in the actual arguments. (like `[ ['vaultName', 'secretPath', 'value'] ]`). Perhaps the naming being 'pair' is kind of misleading. How would I structure this in a map?

Do you mean doing something like putting vault name as the key, and all the provided paths as the values, then use that for the checks here?

Reply:
```ts
const set = new Set<string>();

// The options.preserveNewline fills the set.
for (const [vaultName, path] of options.preserveNewLine) {
  set.add(`${vaultName}${path != null ? `:${path}` : ``}`)
}

// Then when checking you can check the path or the vault name.
function checkPreserveNewline(vaultName: string, path: string): boolean {
  const preserveVault = set.has(vaultName);
  const preservePath = set.has(`${vaultName}:${path}`);
  return preserveVault || preservePath;
}

for (const [vaultName, path] of secrets) {
  if (checkPreserveNewline(vaultName, path)) {
    // Do the thing.
  }
}
```

Can you guess why this would be preferred?